src/DotNetWorker.Core/Hosting/ServiceCollectionExtensions.cs

```diff
@@ -41,52 +41,47 @@ public static IFunctionsWorkerApplicationBuilder AddFunctionsWorkerCore(this ISe
+            services.TryAddSingleton<IFunctionsApplication, FunctionsApplication>();
 
             // Execution
-            services.AddSingleton<IMethodInfoLocator, DefaultMethodInfoLocator>();
-            services.AddSingleton<IFunctionInvokerFactory, DefaultFunctionInvokerFactory>();
-            services.AddSingleton<IMethodInvokerFactory, DefaultMethodInvokerFactory>();
-            services.AddSingleton<IFunctionActivator, DefaultFunctionActivator>();
-            services.AddSingleton<IFunctionExecutor, DefaultFunctionExecutor>();
+            services.TryAddSingleton<IMethodInfoLocator, DefaultMethodInfoLocator>();
+            services.TryAddSingleton<IFunctionInvokerFactory, DefaultFunctionInvokerFactory>();
+            services.TryAddSingleton<IMethodInvokerFactory, DefaultMethodInvokerFactory>();
+            services.TryAddSingleton<IFunctionActivator, DefaultFunctionActivator>();
+            services.TryAddSingleton<IFunctionExecutor, DefaultFunctionExecutor>();
 
             // Function Execution Contexts
-            services.AddSingleton<IFunctionContextFactory, DefaultFunctionContextFactory>();
+            services.TryAddSingleton<IFunctionContextFactory, DefaultFunctionContextFactory>();
 
             // Invocation Features
             services.TryAddSingleton<IInvocationFeaturesFactory, DefaultInvocationFeaturesFactory>();
-            services.AddSingleton<IInvocationFeatureProvider, DefaultBindingFeatureProvider>();
+            services.TryAddSingleton<IInvocationFeatureProvider, DefaultBindingFeatureProvider>();
 
             // Input conversion feature
-            services.AddSingleton<IConverterContextFactory, DefaultConverterContextFactory>();
-            services.AddSingleton<IInputConversionFeatureProvider, DefaultInputConversionFeatureProvider>();
-            services.AddSingleton<IInputConverterProvider, DefaultInputConverterProvider>();
+            services.TryAddSingleton<IConverterContextFactory, DefaultConverterContextFactory>();
+            services.TryAddSingleton<IInputConversionFeatureProvider, DefaultInputConversionFeatureProvider>();
+            services.TryAddSingleton<IInputConverterProvider, DefaultInputConverterProvider>();
 
             // Input binding cache
-            services.AddScoped<IBindingCache<ConversionResult>, DefaultBindingCache<ConversionResult>>();
+            services.TryAddScoped<IBindingCache<ConversionResult>, DefaultBindingCache<ConversionResult>>();
 
             // Output Bindings
-            services.AddSingleton<IOutputBindingsInfoProvider, DefaultOutputBindingsInfoProvider>();
+            services.TryAddSingleton<IOutputBindingsInfoProvider, DefaultOutputBindingsInfoProvider>();
 
             // Worker initialization service
-            services.AddSingleton<IHostedService, WorkerHostedService>();
+            services.TryAddSingleton<IHostedService, WorkerHostedService>();
```

Comment:
This is part of the reason this hadn't been flagged as ready.
I'm still going through all the registrations after the initial update, but this would only succeed if this was the first `IHostedService` registered in the collection.

Updated for now, but I'd like to have a test in place to catch a scenario like this as well.